### PR TITLE
Excel report creation

### PR DESCRIPTION
**Pull request by FrancisRenaud**

### CODE REVIEW

**Comment by Magnitus-** [comment]
@latch2112 

This will require a version bump (as will all code changes in the future which impact shared environments: qa, production, etc).

Semver specifies how to version with api contracts (ie, major increase for backward incompatible changes, minor increase for new feature that doesn't break backward compatibility, bug increase for bug fixes) which seems a bit nebulous for the frontend as the consummer is the end user.

However, we can look at the frontend versioning in terms of the contract it has with the services it consumes. From such a perspective, the new frontend resulting from this PR will break (ie, backward incompatible change) unless we add a new endpoint in the clin proxy api services so a major version change to 2.0.0 would be justified.

@FrancisRenaud

**Comment by latch2112** [comment]
@Magnitus- Thanks for your comment, when can see from it that you have a magnificent knowledge of Semver BUT why did you not took also the opportunities to explain to Francis how to update the version ?

**Comment by FrancisRenaud** [comment]
@latch2112 The point of the above comment was to kindly start the discussion with you on the matter of the choice of proper version number. Now that the discussion has started, do you have a different opinion to offer?

File: client/src/components/screens/PatientVariant/index.jsx [hunk 1]
```diff
@@ -59,6 +62,116 @@ const COLUMN_WIDTHS = {
...
+
+  if (variant.frequencies.gnomAD_exomes) {
+    return getValue('AC')(variant.frequencies.gnomAD_exomes);
+  }
+
+  if (variant.frequencies.gnomAD_genomes) {
+    return getValue('AC')(variant.frequencies.gnomAD_genomes);
+  }
+
+  return '';
```
[the remark this answers is not in the document]
Comment (FrancisRenaud) [reply client/src/components/screens/PatientVariant/index.jsx:114]: Plus court, en effet. Le linter n'aime pas les `else if`, après un `return`, toutefois, puisqu'il les juge non nécessaires.

File: client/src/components/screens/PatientVariant/index.jsx [hunk 2]
```diff
@@ -881,6 +1088,7 @@ class PatientVariantScreen extends React.Component {
       });
     }
 
+    // TODO: optimize this takes around 70 ms to compute
```
[the remark this answers is not in the document]
Comment (FrancisRenaud) [reply client/src/components/screens/PatientVariant/index.jsx:1091]: This is something that will have to be addressed alongside a more generalized performance problem : the whole cycle from `setState` to the end of rendering the page takes `360 ms` on my machine, which means this the minimum delay the user experiences after checking a checkbox before getting the visual feedback for the action. A severe annoyance.

File: client/src/components/screens/PatientVariant/index.jsx [hunk 3]
```diff
@@ -95,10 +208,43 @@ class PatientVariantScreen extends React.Component {
...
+                  <input
+                    type="checkbox"
+                    id={data.mutationId}
+                    value={data.mutationId}
+                    checked={!!selectedVariants[data.mutationId]}
+                    onChange={this.handleSelectVariant}
+                  />
+                );
+              } catch (e) {
+                console.log('Exception while trying to render checkbox: ', e);
```
[the remark this answers is not in the document]
Comment (FrancisRenaud) [reply client/src/components/screens/PatientVariant/index.jsx:240]: oups!

**Pull request merged**